addons/l10n_sa_edi/models/account_move.py

```diff
@@ -32,25 +32,33 @@ def _l10n_sa_is_simplified(self):
         return self.partner_id.company_type == 'person'
 
     @api.depends('amount_total_signed', 'amount_tax_signed', 'l10n_sa_confirmation_datetime', 'company_id',
-                 'company_id.vat', 'journal_id', 'journal_id.l10n_sa_production_csid_json',
-                 'l10n_sa_invoice_signature', 'l10n_sa_chain_index')
+                 'company_id.vat', 'journal_id', 'journal_id.l10n_sa_production_csid_json', 'edi_document_ids',
+                 'l10n_sa_invoice_signature', 'l10n_sa_chain_index', 'state')
     def _compute_qr_code_str(self):
         """ Override to update QR code generation in accordance with ZATCA Phase 2"""
+        phase_one_moves = self.env['account.move']
         for move in self:
-            move.l10n_sa_qr_code_str = ''
-            if move.country_code == 'SA' and move.move_type in ('out_invoice', 'out_refund') and move.l10n_sa_chain_index:
-                edi_format = self.env.ref('l10n_sa_edi.edi_sa_zatca')
-                zatca_document = move.edi_document_ids.filtered(lambda d: d.edi_format_id == edi_format)
+            zatca_document = move.edi_document_ids.filtered(lambda d: d.edi_format_id.code == 'sa_zatca')
+            if move.country_code == 'SA' and move.move_type in ('out_invoice', 'out_refund') and zatca_document and move.state != 'draft':
```

Comment:
It is weird because chain index is actually even a better measure to know whether it has been phase1/2  (and then the only thing missing was the else:)